allow adding annotations for priority-expander configmap

harshagv opened the pull request:
_Description:_

-allow adding annotations for `cluster-autoscaler-priority-expander` configmap

This will allow us to avoid this configmap getting versioned by CD platforms like Spinnaker.

Since any user this CA helm chart using Spinnaker, it adds versioning to the `cluster-autoscaler-priority-expander` like
`cluster-priority-expander-configmap-v01`, which causes CA not to detect this configmap and causing cluster not to scale based on the defined priority. Refer [here](https://github.com/spinnaker/spinnaker/issues/6354).

Slack Conversation: https://kubernetes.slack.com/archives/C09R1LV8S/p1626891316321800?thread_ts=1624547141.173900&cid=C09R1LV8S

harshagv commented:
/assign @gjtempleton

harshagv commented:
/assign @bskiba

gjtempleton commented:
Thanks for the PR!

There's a couple of issues that you'll need to address that the Github workflows which haven't run yet will highlight:
* You'll need to bump the chart version in https://github.com/kubernetes/autoscaler/blob/master/charts/cluster-autoscaler/Chart.yaml#L20
* You'll need to run `helm-docs charts/cluster-autoscaler` from the root of the repo to update the readme with your new variable. Please also cherry-pick in this change at the same time: https://github.com/kubernetes/autoscaler/pull/4208/commits/38e77bec8263aae7ac937cc2562a02461c4b2f0f Helm-docs can be installed following the instructions here: https://github.com/norwoodj/helm-docs#installation


@bskiba Do you have write permissions to the repo to allow running of the pending GH workflows as per [the instructions](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks#approving-workflow-runs-on-a-pull-request-from-a-public-fork)?

gjtempleton commented on charts/cluster-autoscaler/values.yaml:
```diff
@@ -125,6 +125,9 @@ dnsPolicy: ClusterFirst
 expanderPriorities: {}
 
+# priorityConfigMapAnnotations -- Annotations to add to `cluster-autoscaler-priority-expander` ConfigMap.
+priorityConfigMapAnnotations: {}
+
```
Can you add a commented out example for users here please?

harshagv replied:
added in ba47d03

harshagv commented:
> Thanks for the PR!
> 
> There's a couple of issues that you'll need to address that the Github workflows which haven't run yet will highlight:
> 
> * You'll need to bump the chart version in https://github.com/kubernetes/autoscaler/blob/master/charts/cluster-autoscaler/Chart.yaml#L20
> * You'll need to run `helm-docs charts/cluster-autoscaler` from the root of the repo to update the readme with your new variable. Please also cherry-pick in this change at the same time: [38e77be](https://github.com/kubernetes/autoscaler/commit/38e77bec8263aae7ac937cc2562a02461c4b2f0f) Helm-docs can be installed following the instructions here: https://github.com/norwoodj/helm-docs#installation
> 
> @bskiba Do you have write permissions to the repo to allow running of the pending GH workflows as per [the instructions](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks#approving-workflow-runs-on-a-pull-request-from-a-public-fork)?

Added these suggested changes in c32aead

gjtempleton commented on charts/cluster-autoscaler/README.md.gotmpl:
```diff
@@ -327,3 +333,4 @@ Containers:
 Though enough for the majority of installations, the default PodSecurityPolicy _could_ be too restrictive depending on the specifics of your release. Please make sure to check that the template fits with any customizations made or disable it by setting `rbac.pspEnabled` to `false`.
 
 {{ template "chart.valuesSection" . }}
+{"mode":"full","isActive":false}
```
Not sure where this is coming from, but this should be removed.

gjtempleton commented:
Hey, since #4226 got merged and pulled in the cherry-pick I pointed you at, this can actually remove that and you'll need to bump the minor version again after a rebase.

harshagv commented:
> Hey, since #4226 got merged and pulled in the cherry-pick I pointed you at, this can actually remove that and you'll need to bump the minor version again after a rebase.

@gjtempleton thanks for your suggestion, I have added the relevant changes in 97a63e8

harshagv commented:
/assign @gjtempleton

/assign @bskiba

gjtempleton commented:
Thanks for addressing all of the feedback, just squash the commits down to 1 and this will be good to merge!

harshagv commented:
> Thanks for addressing all of the feedback, just squash the commits down to 1 and this will be good to merge!

@gjtempleton  Thanks for your suggestion again. I have squashed all commits using commit ID e605d75

harshagv commented:
> Thanks for addressing all of the feedback, just squash the commits down to 1 and this will be good to merge!

@gjtempleton  Thanks for your suggestion again. I have squashed all commits using commit ID 2dd92cb

gjtempleton commented:
Thanks for that.

/lgtm
/approve
